src/main/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -468,6 +468,36 @@ public SAMFileWriter makeWriter(final SAMFileHeader header, final boolean presor
      *
      */
     public SAMFileWriter makeWriter(final SAMFileHeader header, final boolean presorted, final Path outputPath, final File referenceFasta) {
+        return makeWriter(header, presorted, outputPath, null == referenceFasta ? null : referenceFasta.toPath());
+    }
+
+    /**
+     *
+     * Create a SAM, BAM or CRAM writer based on examination of the outputFile extension.
+     *
+     * @param header header. Sort order is determined by the sortOrder property of this arg.
+     * @param presorted if true, SAMRecords must be added to the SAMFileWriter in order that agrees with header.sortOrder.
+     * @param outputFile where to write the output.  Must end with .sam, .bam or .cram.
+     * @param referenceFasta reference sequence file
+     * @return SAMFileWriter appropriate for the file type specified in outputFile
+     *
+     */
+    public SAMFileWriter makeWriter(final SAMFileHeader header, final boolean presorted, final File outputFile, final Path referenceFasta) {
```

Comment:
Do we really want to add another mixed file/path one?  It seems unnecessary unless I'm missing a use case.

Reply:
Removed.